app/src/main/java/com/futurice/freesound/feature/search/SearchState.java

```diff
@@ -0,0 +1,69 @@
+package com.futurice.freesound.feature.search;
+
+import com.futurice.freesound.common.utils.Preconditions;
+
+import android.support.annotation.NonNull;
+
+import polanski.option.Option;
+
+import static com.futurice.freesound.common.utils.Preconditions.get;
+
+final class SearchState {
```

Comment:
This should be an AutoValue class - that's how you get equals hashCode etc for free.